network-contracts/contracts/KeycardWallet.sol

```diff
@@ -50,7 +51,8 @@ contract KeycardWallet {
     emit TopUp(msg.sender, msg.value);
   }
 
-  constructor(address _owner, address _keycard, Settings memory _settings, address _register) public {
+  constructor(address _owner, address _keycard, Settings memory _settings, address _register,
+      address _optToken, uint256 _optTokenMaxTxAmount) public {
```

Comment:
can't we put this new setting in Setting?

Reply:
no because this goes in the general token mapping. It is a mapping because multiple tokens are allowed in the contact. I wouldn't remove this feature because inside a network there might be some merchants providing their own rewards coin or similar things and at the moment we have a whitelist approach so all allowed tokens must be listed explicitly

Reply:
to add to the previous comment, was your suggestion to put the entire mapping into Setting? It makes sense, but can we send an entire mapping in a tx efficiently even if it is relatively small?